programs/system/src/lib.rs

```diff
@@ -2,33 +2,5 @@
 pub mod system_instruction;
 pub mod system_processor;
 
+use solana_sdk_ids::system_program;
 pub use system_program::id;
-use {
-    solana_account::{state_traits::StateMut, AccountSharedData, ReadableAccount},
-    solana_nonce as nonce,
-    solana_sdk_ids::system_program,
-};
-
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
-pub enum SystemAccountKind {
-    System,
-    Nonce,
-}
-
-pub fn get_system_account_kind(account: &AccountSharedData) -> Option<SystemAccountKind> {
```

Comment:
nit: Although this crate isn't part of the semver policy, this is a breaking change.

If you're worried about downstream breakage, you could re-export the functions from `solana-nonce-account`. Is that a concern? If not, this is fine as is

Reply:
I think you meant that we should re-export from `solana-system-program` crate. I can add that to this PR.

Reply:
Correct, I was suggesting that here we re-export the functions originating from `solana-nonce-account`.

Reply:
The new commit should address this.